git-sync/main.go

```diff
@@ -213,3 +221,27 @@ func setupGitAuth(username, password, gitURL string) error {
 
 	return nil
 }
+
+func setupGitSSH() error {
+	log.Println("setting up git SSH credentials")
+
+	if _, err := os.Stat("/etc/git-secret/id-rsa"); err != nil {
+		log.Fatal("error: could not find SSH secret at /etc/git-secret/id-rsa")
+	}
+
+	if err := os.Mkdir("/root/.ssh", 0600); err != nil {
+		log.Fatal("error creating /root/.ssh directory: %v", err)
+	}
+	config := []byte("IdentityFile /etc/git-secret/id-rsa\nUserKnownHostsFile /dev/null\nStrictHostKeyChecking no\n")
```

Comment:
Kubernetes people upstream may not like this line (disables checking against known hosts)